### ISSUE TITLE
✅ 🔗 🚀 Feature Requests: Add Filter Bar to Posts widget (Merged)

### ISSUE THREAD

**Issue by Isabellesalas** [elementor/elementor#3140]
Hello guys! Is it possible for you to create the option to activate a filter bar on Posts widget, just like the one on the Portfolio widget? Or maybe a pagination feature on Portfolio widget? 'Cause I really miss the possibility to use both of them at once on my blog, pagination and filter bar features.

Best regards

**Comment by harunalikadic** [comment]
+1 for filter bar on Post Widget pages

**Comment by Tama-iris** [comment]
Please filter bar on Post Widget pages or Subtitles under image in portfolio widget. 
Thanks!

**Comment by chocobochick** [comment]
+1 Post Widget filter

**Comment by black-eye** [comment]
+1

**Comment by slomeli79** [comment]
+100 Yes! This request was prematurely closed last year, too.  (https://github.com/pojome/elementor/issues/1716) 

Pro users need this, and it's astounding that it hasn't been added when most competitors have it.

**Comment by ShadowXVII** [comment]
Probably should be tagged with Pro.

**Comment by joelpaul1** [comment]
+1

**Comment by VenkmanNL** [comment]
+1

**Comment by janhenrydesign** [comment]
+1 I was literally just looking for this feature

**Comment by Ramshackles** [comment]
+1 Just needed this aswell

**Comment by stebla27** [comment]
+1
Very useful if you want to design a team member page where you can filter by team member role or location.

**Comment by dasigna** [comment]
+1

**Comment by GwendleWeb** [comment]
+1 This is a must have feature surely and as it's already on the portfolio widget it seems like it should be simple to enable it on the posts widget too.  Please please add this!!

**Comment by Turanligil** [comment]
+1

**Comment by Turanligil** [comment]
+1

**Comment by geebee100** [comment]
+1 Urgent, please, please, please

**Comment by AntonKing60** [comment]
I'm not sure why genuine requests such as this seem to be ignored. They are posted in the appropriate place, they have been clearly identified as a 'missing' element to a widget and yet the guys from Elementor seem to just keep closing the requests as though they have been met. Why would this be?
My search for this very feature/improvement led me here so, can I please add my request to those above for: A Filter bar for Posts (as found in Portfolio) .
Seriously, what's the point in Elementors presence on Github if such requests are just ignored?

**Comment by hbodell** [comment]
+111111111111

**Comment by artconic** [comment]
+111111111111

**Comment by chimayko** [comment]
+1. An absolute must. Let us also filter CPTs using custom taxonomies.

**Comment by btrfabrications** [comment]
Over a year since this request was made (and even older on other (closed!?) requests), and still not a feature? Come on Elementor, please! There surely is no downside to this, and the functionality already exists for portfolios...so why not posts? Filtering is every bit as relevant for posts as it is for any other post type.

@bainternet any news on this, please?

**Comment by itslars** [comment]
+1

**Comment by slomeli79** [comment]
Tried sharing about this neglected feature in a comment thread in the FB user's group, and the admins deleted my comment—not once, but twice. (I thought the first was possibly a mistake, or I'd not hit Enter or something... nope, they removed it both times.)

In reply to a post asking "[Which Elementor add-on you can't live without and why?](https://www.facebook.com/groups/Elementors/permalink/1845057802286664/)", I am pasting my comment here: 

I find myself using Namogo Elementor Extras most frequently, specifically because of the posts-widget filtering capability it offers. This is a feature that Elementor has continued to neglect -

https://github.com/pojome/elementor/issues/3140

https://github.com/pojome/elementor/issues/1716

https://github.com/pojome/elementor/issues/5916

**Comment by Dah19** [comment]
I found this post widget filter bar feature included from a 3rd party add-on listed on Elementor site. To get this enhancement I would have to spend more on a subscription from the third party than I did on Elementor Pro. I also have to try it out to discover what it can and cannot do. I don't think it is a good product strategy for Elementor to leave widget enhancements to third parties that cost as much or more than Elementor pro.  They need to work on a better partner strategy.

**Comment by danielporto** [comment]
> I found this post widget filter bar feature included from a 3rd party add-on listed on Elementor site. To get this enhancement I would have to spend more on a subscription from the third party than I did on Elementor Pro. I also have to try it out to discover what it can and cannot do. I don't think it is a good product strategy for Elementor to leave widget enhancements to third parties that cost as much or more than Elementor pro. They need to work on a better partner strategy. Subscribing to partner bundles to get one plugin is not a good approach.

Can you say which 3rd party addon was that?
and sure, elementor model and the way it handles issues is annoying.
I'm no longer recommending it to anybody. Nor will I be using it anywhere else.

**Comment by Dah19** [comment]
Sure, Element Pack or Essential Add-ons are both listed under third party add-ons.  Element Pack is cheaper.  Good luck.  I did not get a warm fuzzy to see a simple enhancement request from 2017 ignored.  Would think anything to do with blog functionality would be a top priority.  Don't have time to search, try, buy and return plugins that enhance one area and degrade or change what I expected to see in another area.

**Comment by Ramshackles** [comment]
Still missing this feature a lot 👍

**Comment by dasigna** [comment]
dont want to discuss the positives of such native features from scratch... but simply want to point out this is crucial. why are such simple things kept disregarded??

**Comment by lanzalf** [comment]
+1 on the Posts filter. The ability for the user to filter the posts is super important.  I am designing a site with 100s of posts and a filter will be absolutely essential.  Please put this into your future development plan :)

**Comment by janesaastad** [comment]
+1

**Comment by Dah19** [comment]
Elementor is nice.  But it is a tough sell to any customer to justify buying pro and then telling them you need to spend more on third party bundles.  Customers can just sign up for Squarespace or Wix.  They really need to find a better third party strategy.  Would make more sense to sell them through thier own store and control the features roadmap.  To buy a third party bundle to get one widget with one added feature and at the same time loose other features compared to the same Elementor widget does not make business sense for Elementor or customer.

**Comment by andrijanasv** [comment]
Another vote for this feature!!!

**Comment by simonclay** [comment]
Repeatedly come up against needing this feature.

+1

**Comment by btrfabrications** [comment]
Could a developer maybe explain why this hasn't been implemented already? There have been multiple requests across various threads, all asking for the same thing; functionality which already exists in Elementor for the portfolio widget to be included in the posts widget. It seems so simple - what's stopping it? @shilo-ey

**Comment by NicolaCavallazzi** [comment]
+1
And I quote @btrfabrications : they already implemented it in the portfolio, porting the functionality to the posts widget really seems obvious...

**Comment by dicrescenzo3** [comment]
+1
Or I would say o add a card-style option to the portfolio widget to show posts like in posts widget.

**Comment by rolffresh** [comment]
+1

**Comment by NicolaCavallazzi** [comment]
If anyone interested: I wrote a tiny plugin that adds a filter bar to the post widget. You can give it a try [here](https://github.com/NicolaCavallazzi/elementor-super-cat)
Not super straightforward, but if you follow the Readme you should be fine. If you need help ring a bell 😸

**Comment by janesaastad** [comment]
I would love to have this, but I cant figure it out. Is it for downloading?
-Jane

**Comment by NicolaCavallazzi** [comment]
It is! Click on the green button (nearly-top-right) "Clone or Download", download Zip and upload to Wordpress plugins as it is :D

**Comment by janesaastad** [comment]
I have tried many times, it shows as if it downloads but there is nothing in my downloads :-)
-Jane

**Comment by janesaastad** [comment]
Hurra it worked :-)

**Comment by janesaastad** [comment]
Just an important question. How do I find the CSS ID of the post widget? I am an adult lady so things take time = TTT :-)

Thanks!

**Comment by simonclay** [comment]
> If anyone interested: I wrote a tiny plugin that adds a filter bar to the post widget. You can give it a try [here](https://github.com/NicolaCavallazzi/elementor-super-cat)
> Not super straightforward, but if you follow the Readme you should be fine. If you need help ring a bell 😸
> (For a working preview go [here](https://www.monica.studio-pagano.com/blog/))

AMAZING WORK! The filtering is so fast! Thank you! 

Now I want to try to figure out how to get the animation where they 're-shuffle'.

**Comment by NicolaCavallazzi** [comment]
@simonclay Thank you! Maybe I'll consider working on it to get more "stylish" options 😸 
@janesaastad You have to set in manually: just select the Post widget and you should find `CSS ID` in the Advanced tab. Them put the same value in the Post Filter Bar ;)

**Comment by janesaastad** [comment]
hopefully my last question. How do I separate taxonomy in the "Name of taxonomy to filter" 🙄

**Comment by NicolaCavallazzi** [comment]
@janesaastad Didn't think about putting more than 1 taxonomy. I didn't need it. Maybe I'll add this functionality to some future update. At the moment as a workaround you could just put 2 filter bars on the same post widget.

**Comment by rolffresh** [comment]
@NicolaCavallazzi Thanks, great effort. I downloaded and installed, went fine. But the widgets dont show up (elementor pro installed, latest version, with generatepress). I also tried deactivating / activating Elementor Pro.
Any hint? 
thanks

**Comment by stefanico** [comment]
Such a great solution and such a great disappointment on a basic function, I cannot understand why it is still not addressed and sorted... sigh
+999999999999999999999999

**Comment by stefanico** [comment]
A big thanks to @NicolaCavallazzi , I did end up using your plug and it works ... take that elementor dev team ... Still a Pro option is a must have

**Comment by MattOndo** [comment]
+1 this is must have! This request is almost a year old, and a standard for other page builders. Where is it?!

**Comment by Luxamman** [comment]
Yes, this would be very helpfull. 
@bainternet

**Comment by Simongrosman** [comment]
Still nothing?
 +1!

**Comment by typometre** [comment]
+1

**Comment by chrisdt** [comment]
+1

**Comment by Dah19** [comment]
it Would also need to support infinite scrolling, especially important on mobile pages.  Nomago is working on it with filters but has no release date.  the elementor pro widget only offers pagination.  If added filters, you would only see filters for that page I would guess.

**Comment by Dah19** [comment]
it Would also need to support infinite scrolling, especially important on mobile pages. Nomago is working on this with filters but has no release date. the elementor pro widget only offers pagination. If added filters, you would only see filters for that page I would guess.

**Comment by rrodkin** [comment]
+1 Here too . . . seems like a no-brainer!

**Comment by CreativeNaturalista** [comment]
+1

**Comment by codeorlov** [comment]
+1 for filter bar on Post Widget pages

**Comment by davidofug** [comment]
I almost made a feature request only to find this. What is the way forward developers?

**Comment by lucaskeiser** [comment]
+1 Need this for custom post types

**Comment by PieterT2000** [comment]
1+ for filter bar on Archives. Why hasn't this issue been addressed, since it's already open for 2 years or more? Any response would be hugely appreciated!

**Comment by shaneonabike** [comment]
Agreed! I chose Elementor Pro for the feature set but also noticed that Visual Composer has this functionality. You folks could port an adapt the existing work from @NicolaCavallazzi  (and then maybe give them a free year of subscription too)

**Comment by anderscarlen** [comment]
+1, also agree that this a key feature!

**Comment by danielsaezf** [comment]
Thanks so much @NicolaCavallazzi! I'm trying to make it work with a CPT, but it doesn't work. On the filter bar appears all the categories, but when i click on it, nothing happens...

**Comment by NicolaCavallazzi** [comment]
@danielkatz Could you link the page where you have set it up? 😸

**Comment by danielsaezf** [comment]
> @danielkatz Could you link the page where you have set it up? 😸

Of course! [https://hoygrabo.com/recursos](https://hoygrabo.com/recursos)

**Comment by NicolaCavallazzi** [comment]
@danielkatz Okay, easy: you didn't give the CSS ID to the Posts widget. You have to write it in the Filter (and you did -> "recursos"), but you have to write the same ID in the Posts Widget, and it should work! 😸

**Comment by danielsaezf** [comment]
Yes, i know that, and thats what i did, i'll show you an image.

[Link to image](https://hoygrabo.com/wp-content/uploads/2019/05/Captura-de-pantalla-2019-05-28-a-las-12.17.37.png)

**Comment by NicolaCavallazzi** [comment]
Oups, didn't know about that field, my fault 😅 That's not the CSS ID. The ID you want to set is in the Advanced tab and it should be called "CSS ID"

**Comment by danielsaezf** [comment]
> Oups, didn't know about that field, my fault 😅 That's not the CSS ID. The ID you want to set is in the Advanced tab and it should be called "CSS ID"

Wooooow! That was the issue! Thanks a lot! It works and thats all that i needed! A simple filter bar. A big hug from Spain. :D

**Comment by sininenharka** [comment]
Need help with Elementor SuperCat add-on bug on our customer's website. The page should show different categories that can be clicked, these categories filter the services. So now only null category is displayed twice, and it's obviously not a category. Can the developer or someone else help with this? Thanks!

**Comment by shaneonabike** [comment]
@sininenharka  you should really write on their github issue queue where you downloaded the plugin. But to answer your question in a short way you just need to re-edit the widget and select the appropriate fields. I recent fix that they did causes the null values.

**Comment by jasongraphics** [comment]
What the heck Elementor! Still not available?

**Comment by jasongraphics** [comment]
@NicolaCavallazzi Hi! I have pagination on the post library page and it looks like it only filters on that I am on. is there anyway to filter it for all pages? https://www.cleardent.com/testlibrary
This is the site currently. thank you!

**Comment by jasongraphics** [comment]
@NicolaCavallazzi for example, if i goto page 2 of the pagination and there are none of 'eBook' category, clicking the filter bar won't solve the problem. however it is possible to sort it on the first page.

**Comment by danicotillas** [comment]
> > Oups, didn't know about that field, my fault 😅 That's not the CSS ID. The ID you want to set is in the Advanced tab and it should be called "CSS ID"
> 
> Wooooow! That was the issue! Thanks a lot! It works and thats all that i needed! A simple filter bar. A big hug from Spain. :D

Hey @danielsaezf , could you explain me better what did you do exactly? I´m trying on my side with my ID CSS taken from the CPTaxonomy slug and it doesn´t work. Thanks!

**Comment by m2sahin** [comment]
I need it very much right now

**Comment by papplebee** [comment]
+1

**Comment by IDESIGNSTUDIO** [comment]
+1

**Comment by gra-mir** [comment]
Yes please, +1

**Comment by rodolphebertozzo** [comment]
+1

**Comment by wattan1** [comment]
Can't believe this function is still missing!

**Comment by Ramshackles** [comment]
Yeah it’s a bit frustrating and hilarious this hasn’t been added yet


Den 18. dec. 2019 kl. 17.30 skrev wattan1 <notifications@github.com>:

﻿

Can't believe this function is still missing!

**Comment by Ramshackles** [comment]
Yeah guess a polished base product is not as important as adding new features no one probably asked for in the first place

**Comment by MichaelDarko** [comment]
Would also be great when building an archive.

Option to display custom fields in meta in the Archive posts widget and separate filtering widget would let us build all kinds of complex listings.

**Comment by rodolphebertozzo** [comment]
Is it possible for Pro 2.9 ? There's a lot of people asking for it. @shilo-ey

**Comment by realstephenzhao** [comment]
Any update? I have been waiting so long for this.

**Comment by rodolphebertozzo** [comment]
@matipojo

**Comment by dovidmoshee** [comment]
@shilo-ey @KingYes @matipojo Can we please get at least a response? Something? The pro community feels ignored here. Just tell us if it is in the plans to implement or not. Thank you! Shavua Tov!

**Comment by MichaelDarko** [comment]
> Any update? I have been waiting so long for this.
> 
> I have given up waiting for Elementor official, and instead, I switch to Jetsmartfilters from Crocoblock.

The developers of Jet plugins for Elementor are doing god's work. But it's really frustrating that we have to use 3rd party plugins for basic core functionality while Elementor is mainly releasing small niche features. Shouldn't it be the other way round?

**Comment by keremnymn** [comment]
I just want to create a sidebar with filters on it like this: https://www.education.com/resources/second-grade/?cid=11.215

I didn't know it was going to be this difficult. I've been searching how to do it with Elementor for days! And I've just found out that they haven't implemented such an easy and important thing. Wow!

**Comment by looqmedia** [comment]
I `ve found a workaround for me, maybe it´s good for you , too ? You can see it here: 
https://motorradtechnik.de/aktuelles-events/

1. create your categories
2. create a menu from your cats
3. create some posts 

Elementor
1. create an archive template , let´s say 2 columns :
2. here´s how:  https://www.youtube.com/watch?v=7rDIS1Li4jM
3. use categories as filter
4. insert left column = post widget , insert right column = nav widget ( choose and display your cat menu)

That´s it, not elegant , but works ;-)

**Comment by davidamarek** [comment]
Thanks, when you say 'use categories as filter' do you mean the display conditions of the archive?

**Comment by janesaastad** [comment]
What I would like is to have the « Portfolio Filter Bar» in «Posts».

**Comment by aidan2496** [comment]
+1

**Comment by bkjproductions** [comment]
+1

**Comment by davidamarek** [comment]
+1

**Comment by davidamarek** [comment]
Hi @bainternet do you have an update on this, even in terms of where this feature is on the roadmap? I don't really want to have to buy a 3rd party plugin to do this!

**Comment by gerardreches** [comment]
How hasn't this been implemented after more than two years??? It's just the same feature already in use in the Portfolio block, what the heck, it cannot be difficult.

**Comment by realstephenzhao** [comment]
Just release it like Portfolio widget, Can you?

**Comment by foxtrotgolfer** [comment]
One more vote for the Portfolio Filter Bar in Posts! I'd hate to purchase a plug-in for this same functionality.

**Comment by netropol** [comment]
Another vote for adding a filter bar to Posts widget.

**Comment by twerkema** [comment]
+1

**Comment by mfrendi** [comment]
+1

really need this feature

**Comment by dvoorajak** [comment]
Just asked this in Elementor FB commnity. This feature is really missing...

**Comment by mahersinjary** [comment]
I had the same concern, I wanted to add blur effect on hover so I added this in Custom CSS: 

.elementor-portfolio-item:hover img {
    -webkit-filter: blur(12px);
 filter: blur (12px);
}

**Comment by vladimirtana** [comment]
+1 we need this pls

**Comment by Ljuk67** [comment]
+1 I've needed it countless times and it's hard to find. Don't think there's a reliable 3rd party plugin for that, but even if, I'd rather stick to my Elementor PRO features. Please add it (at least to PRO).

**Comment by realstephenzhao** [comment]
Don't expect Elementor Official would take action to improve the core experience, they're busy in marketing not developing.

Actually this so-called Ajax filter bar is not necessary if you use Nav Menu like WPEXPLORER. Just apply the same template to CPT Taxonomy(Display Condition), then you'll get SEO-friendly slugs for Taxonomy Pages.

![](https://i.imgur.com/GDogT7W.png)

**Comment by chloeford0** [comment]
+1 I don't understand why this is on Portfolio items but not Posts?????!!!

**Comment by BoBella** [comment]
+1

**Comment by Samoreen** [comment]
I vote for this feature too.

**Comment by yewzy** [comment]
+1 Yes. This is very much needed.

**Comment by jeffro84** [comment]
+1 from another Pro user

**Comment by KingGrizzly** [comment]
+1 Copying the filter bar functionality from the portfolio widget over to the posts widget would be very helpful.

**Comment by fabianenzensperger** [comment]
Come on, don't be begged like that … @arielk @shilo-ey 

Such a simple but at the same time infinitely useful function is a long time coming. 😔

**Comment by mackyangeles** [comment]
+1,000,000 please stop making us beg for this feature! Some of us including myself are paying yearly for this. This request was since 2017 and until now no progress? You have implemented it on Gallery???

**Comment by Samoreen** [comment]
Hi,

Meanwhile I'm using [this plugin](https://github.com/dettogatto/elementor-super-cat). Not perfect but it's better than nothing. If a single developer could achieve this, I guess that the big Elementor team will have enough self esteem to implement this feature.

**Comment by Samoreen** [comment]
The only problem with this filter bar is that it doesn't paginate well. This may not be an issue depending on your needs.

**Comment by Samoreen** [comment]
> The only problem with this filter bar is that it doesn't paginate well.

I should be more accurate about this. What I mean is that this filter bar only affects the currently displayed posts. So, if you have say, 20 posts displayed on the first page and if applying the filter makes only 5 posts relevant, page 1 is not re-populated with posts from the next pages. If you go to the next page, it will be filtered as well but will probably also have some holes.

**Comment by frdagostino** [comment]
This would be an awesome feature to have! Many clients requests to be able to filter their posts and I always have to add third party plugins with tons of useless code just to have a simple filter bar. Please add this feature :)

**Comment by nerodigital** [comment]
Astounding this isn't a core feature (t least for Pro!). Not to gripe too much but this is the 2nd much-needed feature I've found missing in Elementor Pro today! And just like this one there are tons of requests for it from over 2 years ago. Not sure my Pro license is worth it tbh. 

@frdagostino can I ask which third party plugins you've used to add this functionality?

Many thanks.

**Comment by gerardreches** [comment]
> Astounding this isn't a core feature (at least for Pro!). Not to gripe too much but this is the 2nd much-needed feature I've found missing in Elementor Pro today! And just like this one there are tons of requests for it from over 2 years ago. Not sure my Pro license is worth it tbh.
> 
> @frdagostino can I ask which third party plugins you've used to add this functionality?
> 
> Many thanks.

The easy way for now without adding extra plugins is to use the navigation menu widget + the posts widget. Create a menu with your post categories and introduce it in your archive layouts. That will work, but the page will reload, it won't be an async filter.

Still, Elementor should really introduce this, as there is no reason for the portfolio widget to have it but not the posts widget.

**Comment by frdagostino** [comment]
> @frdagostino can I ask which third party plugins you've used to add this functionality?

If I recall correctly I used the blog widget of Premium Addons for Elementor which is free, but I was not impressed honestly.

Since then on the two websites that required a filter I removed the additional plugins because I don't like having too many plugins extending each other. As a temporary solution I used the solution gerardreches explained.

I'd really like to see this implemented on the pro version though.

**Comment by ghwoodard** [comment]
I will 100% support this feature. Using the Post widget is clutch for blog post and projects/case studies. If you have a lot of content, having visitors being able to filter what they are looking for is ideal. Let's get this on the roadmap Elementor!

**Comment by SYLondon** [comment]
+1 to category filter bar on posts widget.  What I thought would be a 5 minute job is now a major roadblock in carrying out signed off web designs  :( 

If a filter bar, as seen on the Portfolio widget, could be added, that would be ace!?

**Comment by semmtex** [comment]
+1

**Comment by gerardreches** [comment]
Hey guys!

Those of you who are Pro users, please reach to Elementor's support to ask them for this feature. It seems to be more effective to give them our feedback that way than just asking for it on Github. They are who monthly decide what features and improvements the development team should work on based on the support requests that they got.

**Comment by fabianenzensperger** [comment]
+1000000

**Comment by symplTech** [comment]
why this is not there by default?

**Comment by aderr22** [comment]
+1 

I remember looking for this feature 3 years ago - I came looking again because I assumed by now it would be fixed - kind of crazy it isn't. Is there any update available of any kind?

**Comment by EmerKelly** [comment]
+1

**Comment by cranstantinople** [comment]
Posts and portfolio widgets are 90% the same widget but each is missing a key feature the other has.  It seems like you could just combine them into 1 widget with the features/customization from both, you'd have the very versatile widget.

Portfolio widget has a filter option but posts doesn't.
Posts has the option to always display the title in case there isn't a featured Image but Portfolio does not.

Neither really has great layout options for how to display post Title/Images with vs without featured images.  I like the card style you can create with both but it would be great if you could center the text on the card when there isn't a featured image and show it below the image when there is one.

**Comment by meeuwsen** [comment]
+1!

**Comment by ellegaarddk** [comment]
+1 - why wouldn't it be there?

(I've just searched for and tested three different solutions, that is more or less risky to use ... )

**Comment by EmerKelly** [comment]
+1

**Comment by fabianenzensperger** [comment]
+1 👀

**Comment by fabianenzensperger** [comment]
> Posts and portfolio widgets are 90% the same widget but each is missing a key feature the other has and both are missing some minor customizations.
> 
> Portfolio widget has a filter option but posts doesn't. Posts has the option to always display the title in case there isn't a featured Image but Portfolio does not. Neither really has great layout options for how to display post Title/Images with vs without featured images. I like the card style you can create with both but it would be great if you better customize the text layout on posts with vs without featured images. I can usually fix this on the Post widget with css but it would be nice if the options were there.
> 
> It seems like you could just combine them into 1 widget with the features/customization from both with a few more minor customizations, you'd have the very versatile widget.

You are so right. I hope the "Loop Grid" solve all the problems in future!

**Comment by simonclay** [comment]
+1

**Comment by JEricaM** [comment]
I think that this should definitely be a feature.
+ 1 !

**Comment by jonathansoltero** [comment]
Still waiting for this.

**Comment by bkjproductions** [comment]
+1

**Comment by lrdqs** [comment]
+1

**Comment by rehmnanqureshi** [comment]
+1

**Comment by stefckee** [comment]
+1

**Comment by shimsim** [comment]
+1, this would make the new loop builder a solid replacement for many plugins

**Comment by minkidesign** [comment]
I know it's a drop in the Mariana Trench, but do we think there'll be any movement on this?  Literally need it with every project and hate using the 'extras' plugins now that there's the brilliant loop grid...

**Comment by KingGrizzly** [comment]
To @minkidesign 's point: the pattern of needing to filter post, and custom post, content is present on almost every site we build. Typically, we need to use [Facet WP](https://facetwp.com/) in order to achieve filtering of content by taxonomy. It would be nice if the loop grid supported filtering. We typically see this as left-aligned filters, and less frequently top-aligned filters.

Typical left-aligned filter sidebar container:

<img width="327" alt="Screenshot 2023-07-05 at 10 01 29 AM" src="https://github.com/elementor/elementor/assets/48737505/8b364a47-4594-4ee1-8a56-d4c74c89184d">

Having used Facet WP extensively, filtering can include some complexity which may be hard for Elementor to account for, examples include:

- Should filters be additive or subtractive?
- Should filters be radio buttons, check boxes, select dropdowns, or something else?
- Should there be a reset button?
- Which taxonomies should be included?
- Should there be parent and child filters?
- Should counts be displayed next to filters?
- Should there be a maximum number of filters shown before collapsing to a +/- display?

In conclusion: at a basic level, it would be nice if some simple filters could be packaged up with the loop grid widget. However, something which addresses items like the bullets above would be the inevitable need for fully featured filters and I'm not sure if that makes sense as part of the loop widget. Might need to be a new widget altogether.

**Comment by awebguy99** [comment]
I think this may have been added into beta?
Here's a feature request I added some time ago, with more details of what features a filtering system needs (image/brand filtering and lots more);
https://github.com/orgs/elementor/discussions/21203

Please Elementor, you have a chance to make a fully featured filtering system (which is really needed for **any modern e-commerce store**), please implement all these features.

**Comment by lolomarchal** [comment]
+1

**Comment by jevmedia** [comment]
Quick Idea! Maybe you can make it so that you can chose to insert the number of items inside the taxonomy after the text within ()

**Comment by jgarcianewemage** [comment]
and it only took em 4 years xD

**Comment by fabigrafo** [comment]
> and it only took em 4 years xD

And it’s only alpha. Released gut alpha.

**Comment by shimsim** [comment]
> > and it only took em 4 years xD
> 
> And it’s only alpha. Released but alpha.

and its super basic to the point where its still not really viable for many use cases xD

**Comment by fabigrafo** [comment]
> > > and it only took em 4 years xD
> > 
> > 
> > And it’s only alpha. Released but alpha.
> 
> and its super basic to the point where its still not really viable for many use cases xD

Yeah but there is announced more to come for the filters! :)

**Comment by shimsim** [comment]
> > > > and it only took em 4 years xD
> > > 
> > > 
> > > And it’s only alpha. Released but alpha.
> > 
> > 
> > and its super basic to the point where its still not really viable for many use cases xD
> 
> Yeah but there is announced more to come for the filters! :)

There better be :p

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released as the Taxonomy Filter widget for the Loop Builder with Elementor Pro v3.15.0! 🥳 

You can now place a standalone taxonomy filter widget anywhere on your layout and target one of the Loop Grid widgets on the page. You can also use multiple taxonomy filters.

You can learn more about this new feature here: https://elementor.com/help/taxonomy-filter-widget/

Thank you, everyone, for your feedback and patience! 🙏  

We are just getting started with filters. 🤫 😉 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
